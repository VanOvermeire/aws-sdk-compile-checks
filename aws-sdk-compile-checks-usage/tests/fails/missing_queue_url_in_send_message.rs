use aws_config::BehaviorVersion;
use aws_sdk_compile_checks_macro::required_props;

#[required_props]
async fn do_call() {
    let aws_config = aws_config::load_defaults(BehaviorVersion::latest()).await;
    let _something = -3_i32.abs();
    let sqs_client = aws_sdk_sqs::Client::new(&aws_config);
    let _calculate = 3 + 2;
    sqs_client.send_message()
        .message_body(create_message())
        .send()
        .await
        .expect("Call to succeed");
    let _something = -3_i32.abs();
}

fn create_message<'a>() -> &'a str {
    "some message"
}

fn main() {}
