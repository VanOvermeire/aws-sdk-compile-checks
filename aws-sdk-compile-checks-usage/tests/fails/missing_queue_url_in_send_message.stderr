error: method `send_message` (from sqs) is missing required argument(s): `queue_url`
  --> tests/fails/missing_queue_url_in_send_message.rs:10:16
   |
10 |     sqs_client.send_message()
   |                ^^^^^^^^^^^^
